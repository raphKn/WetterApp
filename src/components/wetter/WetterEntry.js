import React from 'react';
import PropTypes from 'prop-types';
import { Image, StyleSheet, View } from 'react-native';
import WetterDescription from './WetterDescription';

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        height: 120,
        alignItems: 'center',
    },
    image: {
        width: 100,
        height: 100,
        margin: 6,
    },
});

const WetterEntry = props => (
    <View style={styles.container}>
        {(<Image
            style={styles.image}
            source={{
                uri: `data:image/png;base64,${props.wetter.icon}`,
            }}
        />)}
        <WetterDescription category={props.wetter.main} description={props.wetter.description} />
    </View>
);

WetterEntry.propTypes = {
    wetter: PropTypes.shape({
        description: PropTypes.string.isRequired,
        main: PropTypes.string.isRequired,
        icon: PropTypes.string.isRequired,
    }).isRequired,
};

WetterEntry.defaultProps = {};

export default WetterEntry;
