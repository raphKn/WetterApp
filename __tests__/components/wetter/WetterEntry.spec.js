import React from 'react';
import renderer from 'react-test-renderer';
import WetterEntry from '../../../src/components/wetter/WetterEntry';

const sonnewetter = {
    icon: 'SonnenIcon',
    description: 'strahlender Sonnenschein',
    main: 'Sonne',
};
jest.mock('../../../src/components/wetter/WetterEntryContainer');

describe('WetterEntry', () => {
    it('renders  correctly in a snapshot', () => {
        const wrapper = renderer.create(
            <WetterEntry wetter={sonnewetter} />);
        expect(wrapper).toMatchSnapshot();
    });
});
