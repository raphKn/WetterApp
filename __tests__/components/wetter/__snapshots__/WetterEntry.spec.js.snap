// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WetterEntry renders  correctly in a snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 120,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "data:image/png;base64,SonnenIcon",
      }
    }
    style={
      Object {
        "height": 100,
        "margin": 6,
        "width": 100,
      }
    }
  />
  <View
    style={
      Object {
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#212121",
          },
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginVertical": 6,
            "textAlign": "center",
          },
        ]
      }
    >
      Sonne
    </Text>
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#212121",
          },
          Object {
            "fontSize": 14,
            "marginVertical": 4,
            "textAlign": "center",
          },
        ]
      }
    >
      - 
      strahlender Sonnenschein
       -
    </Text>
  </View>
</View>
`;
